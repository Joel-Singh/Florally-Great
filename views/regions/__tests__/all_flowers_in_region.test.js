const path = require('path')
const renderPugToDOM = require(path.join(appRoot, 'views', 'renderPugToDOM'))

test("Rendered flower list is empty", () => {
  const locals = {
    region: {
      name: "regionName"
    },
    flower_list: [],
  };

  const renderedDOM = renderPugToDOM('./views/regions/all_flowers_in_region.pug', locals);

  snapshotList(renderedDOM);
})

test("Rendered flower list has flowers", () => {
  const locals = {
    region: {
      name: "regionName"
    },
    flower_list: [
      createFlowerObj('name1', 'desc1', 'url1'),
      createFlowerObj('name2', 'desc2', 'url2'),
      createFlowerObj('name3', 'desc3', 'url3')
    ],
  };

  const renderedDOM = renderPugToDOM('./views/regions/all_flowers_in_region.pug', locals);

  snapshotList(renderedDOM);
})

function snapshotList(DOM) {
  const list = DOM.querySelector('.list')
  expect(list).toMatchSnapshot()
}

function createFlowerObj(name, description, url) {
  return {
    name,
    description,
    url
  }
}
