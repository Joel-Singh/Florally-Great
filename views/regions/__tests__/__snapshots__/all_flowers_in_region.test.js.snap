// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered flower list has flowers 1`] = `
<ul
  class="list"
>
  <li>
    <h2>
      <a
        href="url1"
      >
        name1
      </a>
    </h2>
    <p>
      desc1
    </p>
  </li>
  <li>
    <h2>
      <a
        href="url2"
      >
        name2
      </a>
    </h2>
    <p>
      desc2
    </p>
  </li>
  <li>
    <h2>
      <a
        href="url3"
      >
        name3
      </a>
    </h2>
    <p>
      desc3
    </p>
  </li>
</ul>
`;

exports[`Rendered flower list is empty 1`] = `
<ul
  class="list"
/>
`;
